common/helper_functions.py

```diff
@@ -0,0 +1,15 @@
+from common.bot_logger import get_logger
+
+logger = get_logger(__name__)
+
+
+async def reply_and_delete(ctx, reply: str):
+    try:
+        await ctx.author.send(reply)
+        await ctx.message.delete()
+    except:
+        logger.error(
+            "Could not reply to/remove old message: 'ctx.message.content=%s' by user: 'ctx.author=%s'",
```

Comment:
I don't think you should log an error when `ctx.message.delete()` is called on a PM. That will result in way too many errors when a lot of communication is in those types of messages.

Reply:
I don't think it will be called that often, as the reply and delete are not very likely to fail. But a warning might be more appropriate if I think about it, as the warning would most likely indicate that we are trying to send an invalid reply.

Reply:
The delete will _always_ fail when it is a response on a PM though, so it is 100% likely it will fail.

Reply:
Yeah didn't think of that, added a check for this.